### PR TITLE
Removed @throws tags from IntegrationTestTrait http methods

### PR DESCRIPTION
**Pull request by othercorey**
Alternative to https://github.com/cakephp/cakephp/pull/14193

### CODE REVIEW

File: src/TestSuite/IntegrationTestTrait.php [hunk 1]
```diff
@@ -478,7 +472,6 @@ public function head($url): void
      *
      * @param string|array $url The URL to request.
      * @return void
-     * @throws \PHPUnit\Framework\Error\Error|\Throwable
```
Comment (dereuromark) [inline src/TestSuite/IntegrationTestTrait.php:481]: But this is not really correct, see my intial Comments on the original pr.

File: src/TestSuite/IntegrationTestTrait.php [hunk 2]
```diff
@@ -509,11 +502,7 @@ protected function _sendRequest($url, $method, $data = []): void
                 $this->_requestSession->write('Flash', $this->_flashMessages);
             }
             $this->_response = $response;
-        } catch (PhpUnitError $e) {
-            throw $e;
-        } catch (DatabaseException $e) {
-            throw $e;
-        } catch (LogicException $e) {
+        } catch (PhpUnitError | DatabaseException | LogicException $e) {
```
Comment (garas) [inline src/TestSuite/IntegrationTestTrait.php:505]: `PHPUnit\Framework\Error\Error as PhpUnitError` only catches PHP errors/warning/notices.

`\PHPUnit\Framework\Exception` class or `\PHPUnit\Exception` interface would be better as it catch all PHPUnit asserts too.

File: src/TestSuite/IntegrationTestTrait.php (hunk 1 above)
Comment (dereuromark) [reply src/TestSuite/IntegrationTestTrait.php:481]: As soon as you disableErrorHandlerMiddleware() you can actually expect the \Throwable at least.

File: src/TestSuite/IntegrationTestTrait.php (hunk 2 above)
Comment (tfont) [reply src/TestSuite/IntegrationTestTrait.php:505]: This is a great approach! Well done.

Comment (markstory) [reply src/TestSuite/IntegrationTestTrait.php:505]: I don't think we want to catch phpunit assertion errors here though.

Comment (garas) [reply src/TestSuite/IntegrationTestTrait.php:505]: I wanted to say "it should re-throw all PHPUnit asserts".

Currently they are catched in `\Throwable` block and rendered by `ExceptionRenderer`.

**Comment by othercorey** [comment]
So, do we need to revert this?

**Pull request merged**